src/plugin/plugin.ts

```diff
@@ -116,26 +119,26 @@ const createNodeView = <
     } as unknown) as FieldNameToField<FDesc>[typeof fieldName];
   });
 
-  const getValuesFromNode = (
-    node: Node,
-    decos: Decoration[] | DecorationSet
-  ) => {
-    // We gather the values from each child as we iterate over the
-    // node, to update the renderer. It's difficult to be typesafe here,
-    // as the Node's name value is loosely typed as `string`, and so we
-    // cannot index into the element `fieldDescription` to discover the appropriate type.
-    const fieldValues: Record<string, unknown> = {};
-    node.forEach((node, offset) => {
-      const fieldName = getFieldNameFromNode(
-        node
-      ) as keyof FieldNameToField<FDesc>;
-      const field = fields[fieldName];
-      field.view.onUpdate(node, offset, decos);
```

Comment:
This `getValuesFromNode` also used to update the field views?

Reply:
IIUC, it's only used for the consumer (which doesn't understand the relationship between node and value) – the fieldView understands this relationship and so can derive the value itself if needed.

Reply:
Sorry, bit unclear on your response there. Just to confirm my understanding, it looks like the function to update field views was previously executed as a side effect of the `getValuesFromNode` function. We have now split it out into a separate function. This a) makes it clearer that it's happening and b) allows us to conditionally call it.

In regards to your reply, I'm slightly confused by the term consumer (it's been a while since I've worked with PME).

Reply:
Ah, gotcha – yes, it was a side effect (albeit in an enclosed function), so nice to eliminate!

Re: naming, agreed – I've created https://github.com/guardian/prosemirror-elements/issues/168 to track.